sdk/storage/storage-blob/review/generated/src/models/index.d.ts

```diff
@@ -0,0 +1,6713 @@
+ */
+export interface DataLakeStorageErrorError {
+    /**
+     * The service error code.
+     */
+    code?: string;
+    /**
+     * The service error message.
+     */
+    message?: string;
+}
+/**
+ * An interface representing DataLakeStorageError.
+ */
+export interface DataLakeStorageError {
+    /**
+     * The service error response object.
+     */
+    error?: DataLakeStorageErrorError;
+}
+/**
+ * An Access policy
+ */
+export interface AccessPolicy {
+    /**
+     * the date-time the policy is active
+     * **NOTE: This entity will be treated as a string instead of a Date because the API can
+     * potentially deal with a higher precision value than what is supported by JavaScript.**
+     */
+    start: string;
+    /**
+     * the date-time the policy expires
+     * **NOTE: This entity will be treated as a string instead of a Date because the API can
+     * potentially deal with a higher precision value than what is supported by JavaScript.**
+     */
+    expiry: string;
+    /**
+     * the permissions for the acl policy
+     */
+    permission: string;
+}
+/**
+ * Properties of a blob
+ */
+export interface BlobProperties {
+    creationTime?: Date;
+    lastModified: Date;
+    etag: string;
+    /**
+     * Size in bytes
+     */
+    contentLength?: number;
+    contentType?: string;
+    contentEncoding?: string;
+    contentLanguage?: string;
+    contentMD5?: Uint8Array;
+    contentDisposition?: string;
+    cacheControl?: string;
+    blobSequenceNumber?: number;
+    /**
+     * Possible values include: 'BlockBlob', 'PageBlob', 'AppendBlob'
+     */
+    blobType?: BlobType;
+    /**
+     * Possible values include: 'locked', 'unlocked'
+     */
+    leaseStatus?: LeaseStatusType;
+    /**
+     * Possible values include: 'available', 'leased', 'expired', 'breaking', 'broken'
+     */
+    leaseState?: LeaseStateType;
+    /**
+     * Possible values include: 'infinite', 'fixed'
+     */
+    leaseDuration?: LeaseDurationType;
+    copyId?: string;
+    /**
+     * Possible values include: 'pending', 'success', 'aborted', 'failed'
+     */
+    copyStatus?: CopyStatusType;
+    copySource?: string;
+    copyProgress?: string;
+    copyCompletionTime?: Date;
+    copyStatusDescription?: string;
+    serverEncrypted?: boolean;
+    incrementalCopy?: boolean;
+    destinationSnapshot?: string;
+    deletedTime?: Date;
+    remainingRetentionDays?: number;
+    /**
+     * Possible values include: 'P4', 'P6', 'P10', 'P15', 'P20', 'P30', 'P40', 'P50', 'P60', 'P70',
+     * 'P80', 'Hot', 'Cool', 'Archive'
+     */
+    accessTier?: AccessTier;
+    accessTierInferred?: boolean;
+    /**
+     * Possible values include: 'rehydrate-pending-to-hot', 'rehydrate-pending-to-cool'
+     */
+    archiveStatus?: ArchiveStatus;
+    customerProvidedKeySha256?: string;
+    accessTierChangeTime?: Date;
+}
+/**
+ * An interface representing BlobMetadata.
+ */
+export interface BlobMetadata {
```

Comment:
Why do we need this type? Why not return JS object with properties corresponding to the metadata?

Reply:
I agree with Krzysztof here. "encrypted" is not part of the service so I'm a bit surprised to see this here.

Reply:
Also, I don't know TS well. Does a BlobMetadata object represent all of the metadata key/values or a single one? Also "property" seems like a bad name.

Reply:
There's [another type for the properties](https://github.com/Azure/azure-sdk-for-js/blob/0fb86d67d4e34209b0d930b0f095431100a5facd/sdk/storage/storage-blob/src/generated/src/models/index.ts#L120). BlobMetadata is a dictionary of key/value pairs. I believe metadata is set by users so they can be arbitrary, while properties are fixed.